woocommerce/woocommerce-android
# Issue/7051 promote linked products

Closes #7051 - I'm the sole Android dev on this project, so I simply requested a review from several co-workers. But only one reviewer is needed.

In this PR, we've added a "promo banner" that displays in product detail letting the user know they can add linked products (upsells and cross-sells) to their product. We only show this when the user creates or updates a product, and only when linked products don't already exist.

Thus is hidden behind a feature flag because analytics and tests still need to be added, and we plan to add an A/B test before release.

To test:
* Change [this line](https://github.com/woocommerce/woocommerce-android/blob/400d7f5c180caa0ff0acdeda45a846e2404c8c98/WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt#L364) to always return false so the promo banner won't be shown just once
* Open product detail for a product without linked products
* Make a change and save it
* Notice the "promo banner" appears

https://user-images.githubusercontent.com/3903757/181518538-76223ad5-dfb2-489f-9a3c-337d5b7699a9.mp4
